### PR TITLE
Upgrade to AWS SDK v3

### PR DESCRIPTION
**Pull request by jyustman**
Hi, this seems to be working. Needs a little attention on package.json and package-lock.json, but hopefully this gets you close enough to wrap it up.

### CODE REVIEW

**Comment by sarcastic-verma** [comment]
Any update on this? Looking forward to getting it merged soon.

**Comment by roeysha** [comment]
@ssut +1

**Comment by ssut** [comment]
I will merge this soon.

**Comment by lamkhuong-nfq** [comment]
@ssut how about this PR? Looking forward to getting it merged soon.

**Comment by lamkn06** [comment]
@ssut how about this PR? Looking forward to getting it merged soon.

File: lib/sqs.service.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import { Inject, Injectable, Logger, LoggerService, OnModuleDestroy, OnModuleInit } from '@nestjs/common';
 import { Consumer } from 'sqs-consumer';
 import { Producer } from 'sqs-producer';
+import { GetQueueAttributesCommand, PurgeQueueCommand, QueueAttributeName } from '@aws-sdk/client-sqs';
+import { SQSClient } from '@aws-sdk/client-sqs';
```
Comment (ivanmaximenko) [inline lib/sqs.service.ts:5]:
```suggestion
import { SQSClient , GetQueueAttributesCommand, PurgeQueueCommand, QueueAttributeName } from '@aws-sdk/client-sqs';
```

**Comment by Sohett** [comment]
Any update ? Something blocking ?

**Comment by ssut** [comment]
I apologize for the delay. I have manually merged this PR (https://github.com/ssut/nestjs-sqs/commit/4cc940e4daa90933f228a4cbdfba28c0b31c7671) as I had to change some more tiny things and wanted to check if this changes won't affect the tests. I am going to release v2.0.0 as soon as possible!

**Comment by ssut** [comment]
I just published v2.0.0: https://github.com/ssut/nestjs-sqs/releases/tag/v2.0.0 (npm: https://www.npmjs.com/package/@ssut/nestjs-sqs/v/2.0.0).

**Comment by Sohett** [comment]
@ssut thanks for the new release. 

I don't understand how you setup the AWS credentials though. 

I read on the AWS doc SDK v3 (and on [bbb/sqs-producer](https://github.com/bbc/sqs-producer#credentials)) that you should just provide a `AWS_ACCESS_KEY_ID` and a `AWS_SECRET_ACCESS_KEY` as `.env` variables. 

Nevertheless it's not working anymore for me 😢 now with this new release. 

```
    const sqsError = new SQSError(message);
                     ^
SQSError: SQS receive message failed: The specified queue does not exist for this wsdl version.
    at toSQSError (/node_modules/sqs-consumer/dist/errors.js:40:22)
    at Consumer.receiveMessage (/node_modules/sqs-consumer/dist/consumer.js:159:43)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    ```

**Comment by alexisbellido** [comment]
@Sohett , are you using AWS CLI on your computer? If so, can you try renaming your ~/.aws directory to ~/.aws-temp and see if the problem persists? I was just seeing a similar error and this is what I did to solve it.

Here's the error I was getting:

```
SQSError: SQS receive message failed: Access to the resource https://sqs.us-east-2.amazonaws.com/ is denied.
    at toSQSError (/my-path-to/node_modules/sqs-consumer/dist/consumer.js:48:22)
    at Consumer.receiveMessage (/my-path-to/node_modules/sqs-consumer/dist/consumer.js:162:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

I noticed this by adding a console.log(this.sqs) to  /my-path-to/node_modules/sqs-consumer/dist/consumer.js:162:19 and noticing that the accessKeyId property was being picked from my ~/.aws/credentials file.

I just started figuring out this so I'd appreciate it if anybody else could confirm.

**Comment by Sohett** [comment]
> @Sohett , are you using AWS CLI on your computer? If so, can you try renaming your ~/.aws directory to ~/.aws-temp and see if the problem persists? I was just seeing a similar error and this is what I did to solve it.
> 
> Here's the error I was getting:
> 
> ```
> SQSError: SQS receive message failed: Access to the resource https://sqs.us-east-2.amazonaws.com/ is denied.
>     at toSQSError (/my-path-to/node_modules/sqs-consumer/dist/consumer.js:48:22)
>     at Consumer.receiveMessage (/my-path-to/node_modules/sqs-consumer/dist/consumer.js:162:19)
>     at processTicksAndRejections (node:internal/process/task_queues:96:5)
> ```
> 
> I noticed this by adding a console.log(this.sqs) to /my-path-to/node_modules/sqs-consumer/dist/consumer.js:162:19 and noticing that the accessKeyId property was being picked from my ~/.aws/credentials file.
> 
> I just started figuring out this so I'd appreciate it if anybody else could confirm.

I am indeed using the `AWS CLI`, but this is not the root cause. 
I just tested your proposed solution and it did not solve it. 

Though you helped me solve it! By doing the same as you (console.log in the `sqs-consumer` node package), I noticed the region was set to ``eu-west-1`, which is the one by default! 

🤦 I was not passing the `AWS_REGION` in my credentials ! 
Working like a charm now.

**Pull request closed without merging**